cpg_workflows/large_cohort/combiner.py

```diff
@@ -1,132 +1,129 @@
+
+
+def run_combiner(
+    output_vds_path: Path,
+    sequencing_type: str,
+    cohort: str,
+    tmp_prfx: str,
+    existing_vds_ids: int | None = None,
+) -> VariantDataset:
+    input_vds_samples: list[str] = []
+    use_genome_default_intervals: bool = False
+    use_exome_default_intervals: bool = False
+
+    existing_vds: list[str] | None = None
+    project_gvcf_paths: list[str]
+
+    cohort_query_results: dict[str, Any] = query(
+        GET_COHORT_QUERY,
+        variables={
+            "cohort": cohort,
+        },
     )
 
-    gvcf_paths = [str(s.gvcf.path) for s in sequencing_groups if s.gvcf]
-    if not gvcf_paths:
-        raise ValueError('No sequencing groups with GVCFs found')
-
-    logging.info(f'Combining {len(sequencing_group_names)} sequencing groups: {", ".join(sequencing_group_names)}')
-
-    check_duplicates(sequencing_group_names)
-    check_duplicates(gvcf_paths)
-
-    combiner = hl.vds.new_combiner(
-        gvcf_paths=gvcf_paths,
-        gvcf_sample_names=sequencing_group_names,
-        # Header must be used with gvcf_sample_names, otherwise gvcf_sample_names
-        # will be ignored. The first gvcf path works fine as a header because it will
-        # be only read until the last line that begins with "#":
-        gvcf_external_header=gvcf_paths[0],
-        output_path=str(out_vds_path),
-        reference_genome='GRCh38',
-        temp_path=str(tmp_prefix),
+    init_batch()
+    # Only support a single input cohort
+    formatted_query_results: list[dict[str, Any]] = _parse_metamist_cohort_output(cohort_query_results)
+
+    if existing_vds_ids:
+        vds_query_results: dict[str, Any] = query(
+            GET_VDS_ANALYSIS_QUERY,
+            variables={
+                "vds_ids": existing_vds_ids,
+            },
+        )
+        existing_vds = _parse_metamist_analysis_output(vds_query_results)
+        input_vds_samples = list(chain(input_vds_samples, *[_get_samples_from_vds(vds) for vds in existing_vds]))
+        # Remove any samples from the query that are already present in a VDS
+        formatted_query_results = [sg for sg in formatted_query_results if sg["id"] not in input_vds_samples]
+
+    project_gvcf_paths = [entry["analyses"][0]["output"] for entry in formatted_query_results]
+
+    if sequencing_type == "genome":
+        use_genome_default_intervals = True
+    elif sequencing_type == "exome":
+        use_exome_default_intervals = True
+
+    combiner: VariantDatasetCombiner = new_combiner(
+        output_path=str(output_vds_path),
+        gvcf_paths=project_gvcf_paths,
+        vds_paths=existing_vds,
+        reference_genome="GRCh38",
+        temp_path=tmp_prfx,
+        use_exome_default_intervals=use_exome_default_intervals,
+        use_genome_default_intervals=use_genome_default_intervals,
         force=True,
-        **params,
     )
+
     combiner.run()
-    return hl.vds.read_vds(str(out_vds_path))
+    return read_vds(str(output_vds_path))
```

Comment:
```suggestion
```

I could be wrong, but returning here doesn't do anything. Same in the current implementation. This is run as a python job in a container that's torn down once the VDS is written, so this return goes nowhere

Reply:
actually that's not entirely true - Hail can try to return values from PythonJobs, which is done through pickling. A whole cohort VDS should absolutely not be pickled for a return ⚠️

Reply:
Fair, was just copying from the previous code. I can remove this `return`

Reply:
Return removed